src/vector.rs

```diff
@@ -283,6 +298,32 @@ where
     {
         self.square_length().sqrt()
     }
+
+    /// Returns this vector projected onto another one.
+    #[inline]
+    pub fn project_onto(&self, onto: Self) -> Self
+    where
+        T: Div<T, Output = T>
+    {
+        onto * (self.dot(onto) / onto.square_length())
```

Comment:
do we want to return an `Option`?

Reply:
I would prefer to provide this as is and let users wrap it if they want something bullet proof, just like Vector3D::normalize assumes the vector isn't nil.